### PR TITLE
Upgrade Infection

### PR DESCRIPTION
**Pull request by villfa**
* Fix CI
* Upgrade Infection
* Fix `box compile` when api.github.com returns a 301 code
* Remove symfony/phpunit-bridge

### CODE REVIEW

**Comment by villfa** [comment]
From the logs I see that PHPUnit 9 is not used. I don't know why yet :/

**Comment by villfa** [comment]
I change the goal of this PR because I think it's too early to upgrade PHPUnit mainly because Infection is incompatible with PHPUnit 9.3 for now.

**Comment by theofidry** [comment]
👌

File: composer.json [hunk 1]
```diff
@@ -67,10 +67,10 @@
     },
     "require-dev": {
         "bamarni/composer-bin-plugin": "^1.3",
-        "infection/infection": "^0.10",
+        "infection/infection": "^0.15",
         "mikey179/vfsstream": "^1.6",
-        "phpunit/phpunit": "^8.1",
-        "symfony/phpunit-bridge": "^4.2 || ^5.0"
```
Comment (theofidry) [inline composer.json:73]: out of curiosity why removing the bridge? I think its deprecation helper is quite helpful

Comment (villfa) [reply composer.json:73]: I thought the library was not used at all. I didn't think about the deprecation helper.
I can make another PR to put it back if you wish.

Comment (theofidry) [reply composer.json:73]: That would be nice 👍

Comment (villfa) [reply composer.json:73]: Done #475

**Pull request merged**